Contributing.md

```diff
@@ -30,10 +30,20 @@ This project uses clang-format (stable branch) to check for common style issues.
 
 In most cases, clang-format can and **should** be used to automatically reformat code and solve most formatting issues.
 
-- To run clang-format on all staged files:
+It can be downloaded as part of the LLVM package here: [LLVM Downloads](http://releases.llvm.org/download.html)
+
+- To run clang-format on all staged files use the following command:
   ```
   git diff --cached --name-only | egrep '[.](cpp|h|mm)$' | xargs clang-format -i
   ```
+  - **Warning:** This method does not take the `.clang-format` configuration file into account, results may vary. While it may perform good enough to pass Lint checks on github, it will not be 100% accurate.
+  
+- Clang-format can also be used inside Visual Studio 2017 with this extension:
```

Comment:
`clang-format`

Also, consider including the link to the plugin directly and in a less redundant way:

> VS2017 users can install the [clang-format plugin](https://marketplace.visualstudio.com/items?itemName=HansWennborg.ClangFormat#overview).

I'm not sure how useful and relevant the following information is. I don't see why we need to mention the .clang-format stuff, the additional options (which are unrelated to this contributing guide) or the default plugin settings (which may change, and then the info here would become outdated).

Reply:
If it turns out that the discrepancy was not about the style settings file, then ofcourse this will be changed.